[Snyk] Upgrade punycode from 2.1.1 to 2.3.1

alikon opened the pull request:
<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br /><h3>Snyk has created this PR to upgrade punycode from 2.1.1 to 2.3.1.</h3>

:information_source: Keep your dependencies up-to-date. This makes it easier to fix existing vulnerabilities and to more quickly identify and fix newly disclosed vulnerabilities when they affect your project.
<hr/>

- The recommended version is **4 versions** ahead of your current version.
- The recommended version was released **4 months ago**, on 2023-10-30.


<details>
<summary><b>Release notes</b></summary>
<br/>
  <details>
    <summary>Package name: <b>punycode</b></summary>
    <ul>
      <li>
        <b>2.3.1</b> - <a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/releases/tag/v2.3.1">2023-10-30</a></br>No content.
      </li>
      <li>
        <b>2.3.0</b> - <a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/releases/tag/v2.3.0">2023-01-19</a></br>No content.
      </li>
      <li>
        <b>2.2.2</b> - <a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/releases/tag/v2.2.2">2023-01-19</a></br>No content.
      </li>
      <li>
        <b>2.2.0</b> - <a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/releases/tag/v2.2.0">2023-01-10</a></br>No content.
      </li>
      <li>
        <b>2.1.1</b> - <a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/releases/tag/v2.1.1">2018-05-22</a></br>No content.
      </li>
    </ul>
    from <a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/releases">punycode GitHub release notes</a>
  </details>
</details>


<details>
  <summary><b>Commit messages</b></summary>
  </br>
  <details>
    <summary>Package name: <b>punycode</b></summary>
    <ul>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/9e1b2cda98d215d3a73fcbfe93c62e021f4ba768">9e1b2cd</a> Release v2.3.1</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/ef3505c8abb5143a00d53ce59077c9f7f4b2ac47">ef3505c</a> Prepare v2.3.1 release</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/f1e682ce3695817d1a0100e14afa94040ea0ddab">f1e682c</a> Update &#x60;version&#x60;</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/1287b463f918c6369c840140df72fa967a3e6d26">1287b46</a> ci(deps): update GitHub Actions workflows to run on Node.js 20 (LTS) (#130)</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/d8bbc1fe1d5f8061ecf46dc4d356f41b0cea7634">d8bbc1f</a> Update dependencies (#128)</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/36db01b15285f364fd5a40df56fdb62241e46bd1">36db01b</a> Fix broken reference</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/1f3c3dc0497312846b160c4e0ad6d6635fef4c8b">1f3c3dc</a> Add README section for maintainers</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/2047fe7e4ddcece709b388a569be51c62045535e">2047fe7</a> Release v2.3.0</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/7786653e242f775794e47ff011495709903edde3">7786653</a> Rename package.json#name before publishing &#x60;punycode.js&#x60;</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/b77c249baff49fadfcc91a7c0abc5d5160dfa6ca">b77c249</a> Release v2.2.2</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/baa6683b38bfe45866e49cda75f2459063af6975">baa6683</a> Update repo URL</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/efd0776fc593a4b4915502b18728f03c276da984">efd0776</a> Add jsDelivr hits badge (#69)</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/533129ebc8effa48f2588c297a638151442ca8eb">533129e</a> Update mocha dependency (#103)</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/81a65d56940043f81a527865954bbbdb3e2324ec">81a65d5</a> Set up GitHub Actions</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/cd1a4baaf8cf0b4e33469024970527b3e63041f9">cd1a4ba</a> Add test for #115</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/73a3fe42946ba03a2bfb2ac71c23913d763224df">73a3fe4</a> Do not encode DEL (#115)</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/02227ad988ce186384c7b6249747f83640e42ce4">02227ad</a> Update browser support section in README (#118)</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/0ea831c9e2ec748201118eeb39f6412510421b15">0ea831c</a> Replace &#x60;let&#x60; with &#x60;const&#x60; where applicable (#93)</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/7c2c9e9c2753d92643caba32f75ab65c7e839be7">7c2c9e9</a> Release v2.2.1</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/7d6b9e62cab3626729874be9da15a46a57b5df9e">7d6b9e6</a> Do not decode non-ASCII-alphanumerics in Punycode labels (#124)</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/bff3e35710e8e6b21983da388b6ad642462e60f7">bff3e35</a> Release v2.2.0</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/3d53cc4f3ed789beef62913e5b2ed3068b3ed996">3d53cc4</a> fix: upstream node.js changes (#121)</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/80ddeaff3f0029c99175f28275d86461fdd1e9fe">80ddeaf</a> fix: update jsdoc definitions (#120)</li>
      <li><a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/commit/6cd1ddd078176a5b2afdf09c60633217e2009e53">6cd1ddd</a> Fix usage instructions in README (#113)</li>
    </ul>

   <a href="https://snyk.io/redirect/github/mathiasbynens/punycode.js/compare/68df855dc42d1086ada161331b3074468e8d848d...9e1b2cda98d215d3a73fcbfe93c62e021f4ba768">Compare</a>
  </details>
</details>
<hr/>

**Note:** *You are seeing this because you or someone else with access to this repository has authorized Snyk to open upgrade PRs.*

For more information:  <img src="https://api.segment.io/v1/pixel/track?data=eyJ3cml0ZUtleSI6InJyWmxZcEdHY2RyTHZsb0lYd0dUcVg4WkFRTnNCOUEwIiwiYW5vbnltb3VzSWQiOiI2NGI0NTY1Yy04ZDBjLTQwNmEtYmM3ZS02OTNhOTEyYWZkYzMiLCJldmVudCI6IlBSIHZpZXdlZCIsInByb3BlcnRpZXMiOnsicHJJZCI6IjY0YjQ1NjVjLThkMGMtNDA2YS1iYzdlLTY5M2E5MTJhZmRjMyJ9fQ==" width="0" height="0"/>

🧐 [View latest project report](https://app.snyk.io/org/alikon/project/7f972e16-a9b0-41f9-a764-0801652e912f?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr)

🛠 [Adjust upgrade PR settings](https://app.snyk.io/org/alikon/project/7f972e16-a9b0-41f9-a764-0801652e912f/settings/integration?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr)

🔕 [Ignore this dependency or unsubscribe from future upgrade PRs](https://app.snyk.io/org/alikon/project/7f972e16-a9b0-41f9-a764-0801652e912f/settings/integration?pkg&#x3D;punycode&amp;utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;upgrade-pr#auto-dep-upgrades)

<!--- (snyk:metadata:{"prId":"64b4565c-8d0c-406a-bc7e-693a912afdc3","prPublicId":"64b4565c-8d0c-406a-bc7e-693a912afdc3","dependencies":[{"name":"punycode","from":"2.1.1","to":"2.3.1"}],"packageManager":"npm","type":"auto","projectUrl":"https://app.snyk.io/org/alikon/project/7f972e16-a9b0-41f9-a764-0801652e912f?utm_source=github&utm_medium=referral&page=upgrade-pr","projectPublicId":"7f972e16-a9b0-41f9-a764-0801652e912f","env":"prod","prType":"upgrade","vulns":[],"issuesToFix":[],"upgrade":[],"upgradeInfo":{"versionsDiff":4,"publishedDate":"2023-10-30T18:28:32.512Z"},"templateVariants":[],"hasFixes":false,"isMajorUpgrade":false,"isBreakingChange":false,"priorityScoreList":[]}) --->

richard67 commented:
I just see that tests are missing for the new function. Could you add tests? OR shall I try?

richard67 commented:
I also noticed that the @since tag in the new function's documentation needs to be changed, but I don't remember the constant text to be added, which will trhen be replaced with the Joomla! version when being released. Do you know that constant text for the @since tag?

snyk-bot committed:
fix: upgrade punycode from 2.1.1 to 2.3.1

Snyk has created this PR to upgrade punycode from 2.1.1 to 2.3.1.

See this package in npm:
https://www.npmjs.com/package/punycode

See this project in Snyk:
https://app.snyk.io/org/alikon/project/7f972e16-a9b0-41f9-a764-0801652e912f?utm_source=github&utm_medium=referral&page=upgrade-pr
